newlib/libc/machine/arm/aeabi_memcpy-armv7a.S

```diff
@@ -242,31 +259,50 @@ src_not_word_aligned:
         bge     9b
```

Comment:
I believe the `ldr` instruction on line 256 just above can also use an unaligned address.
(For some reason GitHub doesn't allow me to add a comment directly on the relevant line)

Reply:
Good catch. Looking at the code this morning, what if we just did a four-byte loop instead of a 64-byte loop for this case? That would make the code a lot smaller, although it would run a bit slower for this unaligned case.

Reply:
Sorry, I missed your reply last week. Yes, I believe that would definitely help with code-size.
I wonder the benefit of having a single `str` instruction for each word is still worth the extra complexity in this case, but I reckon it probably is.

Reply:
It's all a space/speed tradeoff. Maybe the right solution is to move this to C though; it's not like asm is helping much for this case.

Reply:
Looks great! Thanks for the update.